### DISCUSSION TITLE
Regarding legend in postprocessing

### DISCUSSION THREAD

**Discussion by dashp7** [AMReX-Codes/amrex#2349] [Q&A]
Hi,

I am using the [fsnapshot](https://github.com/AMReX-Codes/amrex/blob/development/Tools/Plotfile/fsnapshot.cpp) tool to plot single variable data. I wanted to know:
1. Is it possible to include a legend in the rendering?(currently it prints the max and the min to the console, but a legend would be more helpful)
2. Is there a way to save the rendering directly in png/jpg format?

Thanks.

**Comment by zingale** [comment]
that script is really basic and only supports simple bitmap output.  To do a png we would need to build in a png library, adding a dependency.  Likewise for adding a legend, we would need a graphics library that does text rendering.  I think these are beyond what we really want to do.

You have some options though.  Imagemagick convert allows you to convert the file type to pretty much anything (including PNG) and can also add text to the image, it that's what you want.  You can even use shell syntax to loop over files to batch process a bunch.  If you look at the convert documentation you should see how that works.

But a better long term strategy might be to start using yt for these things (https://yt-project.org/).  There's a little learning curve, but you can easily output PNG and text legends, colorbars, etc. and make many other types of plots.
